lib/active_merchant/billing/gateways/barclaycard_smartpay.rb

```diff
@@ -210,12 +210,18 @@ def headers(account, password)
       end
 
       def parse(response)
-        Hash[
-          response.split('&').map do |x|
-            key, val = x.split('=', 2)
-            [key.split('.').last, CGI.unescape(val)]
+        parsed_response = {}
+        response.split('&').map do |x|
```

Comment:
It's really confusing what this is doing - can't we just use: http://ruby-doc.org/stdlib-2.6.3/libdoc/cgi/rdoc/CGI.html#method-c-parse?

Reply:
good call - updated 👍 Outside of this spot, does the rest of the method seem overly complex (creating the nested hash)?